src/hooks/queries/useCustomValidationConfig.ts

```diff
@@ -0,0 +1,48 @@
+import { useQuery } from '@tanstack/react-query';
+import type { UseQueryResult } from '@tanstack/react-query';
+import type { AxiosError } from 'axios';
+
+import { useAppQueries } from 'src/contexts/appQueriesContext';
+import { CustomValidationActions } from 'src/features/customValidation/customValidationSlice';
+import { useAppDispatch } from 'src/hooks/useAppDispatch';
+import { useAppSelector } from 'src/hooks/useAppSelector';
+import { getCurrentDataTypeForApplication } from 'src/utils/appMetadata';
+import { resolveExpressionValidationConfig } from 'src/utils/validation/expressionValidation';
+import type { IApplicationMetadata } from 'src/features/applicationMetadata';
+import type { ILayoutSets } from 'src/types';
+import type { IInstance } from 'src/types/shared';
+import type { IExpressionValidationConfig } from 'src/utils/validation/types';
+
+export const useCustomValidationConfig = (): UseQueryResult<IExpressionValidationConfig | null> => {
+  const dispatch = useAppDispatch();
+  const { fetchCustomValidationConfig } = useAppQueries();
+
+  const appMetadata: IApplicationMetadata | null = useAppSelector(
+    (state) => state.applicationMetadata.applicationMetadata,
+  );
+  const instance: IInstance | null = useAppSelector((state) => state.instanceData.instance);
+  const layoutSets: ILayoutSets | null = useAppSelector((state) => state.formLayout.layoutsets);
+
+  const dataTypeId =
+    getCurrentDataTypeForApplication({
+      application: appMetadata,
+      instance,
+      layoutSets,
+    }) ?? '';
```

Comment:
I've made a hook for this whole thing, `useCurrentDataModelName()` 🥳

Reply:
Oh nice!